### PR TITLE
Factor out common test code for running `datalad fusefs`

### LINKED ISSUE #58: RF tests to avoid duplication of setting up a fused location to be used
code like 
```python
    CONTENT = "This is test text.\n"
    ds = Dataset(tmp_path / "ds").create(cfg_proc="text2git")
    (tmp_path / "ds" / "text.txt").write_text(CONTENT)
    ds.save(message="Create text file")
    mount = tmp_path / "mount"
    mount.mkdir()
    p = subprocess.Popen(
        [
            "datalad",
            "fusefs",
            "-d",
            ds.path,
            "--foreground",
            "--mode-transparent",
            str(mount),
        ]
    )
    with pytest.raises(subprocess.TimeoutExpired):
        p.wait(timeout=3)
```
is very common to many tests.  IMHO it should be one or two fixtures (or just helper functions or context managers) to be (re)used by tests to establish the source dataset + fuse mount.

### PR DESCRIPTION
**Pull request by jwodder**
Closes #58.

### CODE REVIEW

File: datalad_fuse/tests/test_fuse.py [hunk 1]
```diff
@@ -1,31 +1,60 @@
...
     with pytest.raises(subprocess.TimeoutExpired):
         p.wait(timeout=3)
-    assert sorted(q.name for q in tmp_path.iterdir()) == dots + sorted(data_files)
-    for fname, blob in data_files.items():
-        assert (tmp_path / fname).read_bytes() == blob
-    p.terminate()
+    try:
+        yield mount_dir
+    finally:
+        p.terminate()
```
Comment (yarikoptic) [inline datalad_fuse/tests/test_fuse.py:42]: is that where we could test if that process is still running (in the context of #56 ) -- I mentioned that all those processes remained running until some (5th) was finally to stall

Comment (jwodder) [reply datalad_fuse/tests/test_fuse.py:42]: Yes, we could test there, but I don't think it's needed anymore.  While testing out #65, running `ps` in another window while tests were running showed that the datalad processes were exiting when they were supposed to.

File: datalad_fuse/tests/test_fuse.py [hunk 2]
```diff
@@ -1,31 +1,60 @@
...
     )
     # Check that the command didn't fail immediately:
     with pytest.raises(subprocess.TimeoutExpired):
         p.wait(timeout=3)
-    assert sorted(q.name for q in tmp_path.iterdir()) == dots + sorted(data_files)
-    for fname, blob in data_files.items():
-        assert (tmp_path / fname).read_bytes() == blob
-    p.terminate()
+    try:
+        yield mount_dir
```
Comment (yarikoptic) [inline datalad_fuse/tests/test_fuse.py:40]: Looking at all `dots` which are tightly linked to `transparent`: May be here to return some data struct with `.mount`, `.dots` etc to be used in the tests?  could later even then provide some `.pid` if some test would desire to interrogate it more or smth like that.

Comment (yarikoptic) [reply datalad_fuse/tests/test_fuse.py:40]: oh, nevermind -- `fusing` is not the one responsible for creating the dataset, my bad

File: datalad_fuse/tests/test_fuse.py (hunk 1 above)
Comment (yarikoptic) [reply datalad_fuse/tests/test_fuse.py:42]: Well, ok, although for me it is the point of unit tests to ensure what discovered to be broken now wouldn't need fishing expedition in the future by us or next generation of developers ;-)

**Comment by yarikoptic** [comment]
ok, let's proceed as is for now.  Thank you @jwodder !

**Pull request merged**